### ISSUE TITLE
[CT-721] [Bug] Newlines in multiline description  

### ISSUE THREAD

**Issue by alexnikitchuk** [dbt-labs/dbt-core#5330]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

As per [doc](https://docs.getdbt.com/reference/resource-properties/description#add-a-multiline-description-to-a-model) to add a multiline description to a model we can use YAML [block notation](https://yaml-multiline.info/). According to the notation if we would like to keep newlines as newlines, use the literal style, indicated by a pipe (|). If instead, we want them to be replaced by spaces, use the folded style, indicated by a right angle bracket (>).
Here is an example with literal style, but in dbt docs all newlines are replaced with spaces 

```
version: 2

models:
  - name: a_model
    description: |
      Line #1.
      Line #2.
```

### Expected Behavior

All newlines kept as newlines if we use the literal style, indicated by a pipe (|)

### Environment

```markdown
- OS: macOS 12.4
- Python: 3.8.13
- dbt: 1.1.0
```

### What database are you using dbt with?

snowflake

**Comment by jaypeedevlin** [comment]
Hi Alex.  I don't work for dbt Labs but I looked into this one for fun/procrastination.

Line breaks seem to make it into the manifest:

```json
"source_description": "Line of text\nSingle Linebreak\n\nDouble linebreak\n",
```

[dbt-docs uses angular-marked 1.2.2](https://github.com/dbt-labs/dbt-docs/blob/a805ac815aba45d2173f2c48b4647e5177131a2a/package.json#L10) for markdown rendering, [which in turn uses marked 0.3.3](https://github.com/Hypercubed/angular-marked/blob/v1.2.2/package.json#L10).

[dbt-docs specifies to use gfm](https://github.com/dbt-labs/dbt-docs/blob/a805ac815aba45d2173f2c48b4647e5177131a2a/src/app/index.module.js#L28), but [that's only supported from marked 0.29](https://marked.js.org/). As a result dbt-docs is using strict markdown, where a single line break doesn't trigger a new paragraph:

<img width="455" alt="image" src="https://user-images.githubusercontent.com/17235807/172302905-c00878a1-9f21-4de2-8959-a6f4eb9a5753.png">

TL;DR there seem to be a few possible solutions here:

1. Find a new angular library for dbt-docs that supports gfm
2. Fork angular-marked and bump the marked dependency
3. Use double line breaks in descriptions when you want a new paragraph (shown below).

<img width="258" alt="image" src="https://user-images.githubusercontent.com/17235807/172302978-03f88046-7fcc-4406-af0e-da61b80ed3d9.png">

**Comment by alexnikitchuk** [comment]
thanks for the investigation @jaypeedevlin 
But newline does not work:
<img width="322" alt="image" src="https://user-images.githubusercontent.com/8694829/172325463-94282b1b-50c7-4dd2-9a99-8ce17ffbe1dd.png">
BTW `<br>` worked in v0.21

**Comment by alexnikitchuk** [comment]
yup, this gives us whole new paragraph but not just a new line.
Also model yaml looks overloaded with newlines if we use it:
```
  - name: my_model
    description: >
      Line 1


      Line 2


      Line 3
      

      Line 4
```

**Comment by jaypeedevlin** [comment]
If this is possible currently, I don't know how, unfortunately.  One thing I will point out though is that in your example above you have three line breaks, not two (two line breaks would give you one empty line).

Personally I don't mind this formatting, as it lets me construct YAML like this where I can wrap the lines without affecting the paragraph:

<img width="881" alt="image" src="https://user-images.githubusercontent.com/17235807/172499599-a3f37423-adb2-42de-879b-a50071f6df5a.png">

That said, it seems like the intention in the dbt-docs code is for GFM to be used and it's currently not being (AFAICT) so maybe there's the possibility to resolve this on the dbt side — we can wait for them to weigh in!

**Comment by Gregory108** [comment]
Neither (one) newline, nor `: |`, nor `<br>`, nor `<br/>` work. The issue is there and should be re-opened

**Comment by alexnikitchuk** [comment]
@Gregory108 it works: try to add **two or more spaces and a newline** as I mentioned in the comment you disliked
```
version: 2

models:
  - name: my_model
    description: |
      line 1  
      line 2
```

**Comment by Gregory108** [comment]
@alexnikitchuk I tried it before and wanted to demonstrate it with screenshots now... Which led me to notice my VSCode automatically trims whitespace at the end of the line - which erased my manual whitespace. I revoke my dislike=)

Additional tuning is required in folder (or Workspace or User) settings:
```json
    "[yaml]": {
        "files.trimTrailingWhitespace": false,
    },
```

Still, I stand behind the statement the issue is present: DBT Cloud documentation interprets code not as expected / not intuitive / not as declared in [DBT documentation about `description:` parameter](https://docs.getdbt.com/reference/resource-properties/description#add-a-multiline-description-to-a-model).

PS: we even have a whitespace trimmer built into `pre-commit`, which is useful for all other documentation parameters, but has to be put on hold for this one quirk.

**Comment by hvignolo87** [comment]
> PS: we even have a whitespace trimmer built into `pre-commit`, which is useful for all other documentation parameters, but has to be put on hold for this one quirk:
> 
> ```yaml
> repos:
> -   repo: https://github.com/pre-commit/pre-commit-hooks
>     rev: <your-version>
>     hooks:
>     -   id: trailing-whitespace
>         exclude: .*.yml$
> ```

I'm in the same situation, I'll need to explicitly exclude the dbt YAML files for that hook, which is a workaround for this issue.

Is there any other open issue related to this one? I can't find one 🤔
